tests/functional/api/groups/test_create.py

```diff
@@ -19,6 +19,19 @@ def test_it_returns_http_200_with_valid_payload(self, app, token_auth_header):
 
         assert res.status_code == 200
 
+    @pytest.mark.parametrize(
```

Comment:
I did add functests for all endpoints pertaining to their version support. It did help me catch a couple that almost slipped through the cracks, but these functests may be too slow-y down-y to be useful? I dunno.

Reply:
I do think testing each and every endpoint separately for multiple version support is redundant and that the costs probably outweigh the benefits. I would be inclined to:

- Have an endpoint to check version support for one or a small number of routes
- Have a test to check the index route output for v1 and v2, since I think this is assembled from the same metadata that is used to implement other aspects of versioning support. If you forgot to add "v2" to one of the API routes, presumably it would not show up in the index route output?

Reply:
Yes, let me rethink this. The thing I'm trying to test-protect us for is: inadvertently not having version support for an endpoint. We're protected against having two views trying to provide support for a single version—the opposite problem—because that will raise during config.

Given the way `@api_config` works, all version support can be gleaned from registry entries, I shall endeavor to find a straightforward way to test at this level...